src/WebUI/Components/DaemonSetListingComponent.tsx

```diff
@@ -26,97 +27,94 @@ export class DaemonSetListingComponent extends BaseComponent<IDaemonSetComponent
-            columnActionsMode: ColumnActionsMode.disabled
+            renderCell: (rowIndex: number, columnIndex: number, tableColumn: ITableColumn<V1DaemonSet>, daemonSet: V1DaemonSet) => this._renderAgeCell(rowIndex, columnIndex, tableColumn, daemonSet),
         });
 
-
         return columns;
     }
-    private static _onRenderItemColumn(daemonSet?: V1DaemonSet, index?: number, column?: IColumn): React.ReactNode {
-        if (!daemonSet || !column) {
-            return null;
-        }
 
-        let textToRender: string | undefined;
-        let colDataClassName: string = "list-col-content";
-        switch (column.key) {
-            case setNameKey:
-                return ListComponent.renderTwoLineColumn(daemonSet.metadata.name,
-                                                        Utils.getPipelineText(daemonSet.metadata.annotations),
-                                                        colDataClassName,"primary-text", "secondary-text");
-            case imageKey:
-                textToRender = daemonSet.spec.template.spec.containers[0].image;
-                break;
-            case podsKey: {
-                let statusProps: IStatusProps | undefined;
-                let podString: string = "";
-                if (daemonSet.status.desiredNumberScheduled > 0) {
-                    statusProps = Utils._getPodsStatusProps(daemonSet.status.currentNumberScheduled, daemonSet.status.desiredNumberScheduled);
-                    podString = format("{0}/{1}", daemonSet.status.currentNumberScheduled, daemonSet.status.desiredNumberScheduled);
-                }
-                return (
-                    <div>
-                        {
-                            !!statusProps &&
-                            <Status {...statusProps} size={StatusSize.m} />
-                        }
-                        <span>{podString}</span>
-                    </div>);
-            }
-            case ageKey: {
-                return (<Ago date={new Date(daemonSet.metadata.creationTimestamp)} />);
-            }
+    private static _renderSetNameCell(rowIndex: number, columnIndex: number, tableColumn: ITableColumn<V1DaemonSet>, daemonSet: V1DaemonSet): JSX.Element {
+        return (
+            <SimpleTableCell columnIndex={columnIndex} tableColumn={tableColumn} key={"col-" + columnIndex}>
+                {ListComponent.renderTwoLineColumn(daemonSet.metadata.name,
+                    Utils.getPipelineText(daemonSet.metadata.annotations),
+                    colDataClassName, "primary-text", "secondary-text")}
+            </SimpleTableCell>);
+    }
+
+    private static _renderImageCell(rowIndex: number, columnIndex: number, tableColumn: ITableColumn<V1DaemonSet>, daemonSet: V1DaemonSet): JSX.Element {
+        const textToRender = daemonSet.spec.template.spec.containers[0].image;
+        return (
+            <SimpleTableCell columnIndex={columnIndex} tableColumn={tableColumn} key={"col-" + columnIndex}>
```

Comment:
may be we can push the encapsulating <SimpleTableCell/> to the ListComponent and just specify the inner child components from here. otherwise we are repeating ourselves here.